src/native/fs/fs_napi.cpp

```diff
@@ -1941,6 +1945,53 @@ struct DirOpen : public FSWorker
     }
 };
 
+struct TellDir : public FSWrapWorker<DirWrap>
+{
+    std::string _dir_path;
+    long int _tell_res;
+    TellDir(const Napi::CallbackInfo& info)
+        : FSWrapWorker<DirWrap>(info)
+    {
+        Begin(XSTR() << "TellDir " << DVAL(_wrap->_path));
+    }
+    virtual void Work()
+    {
+        DIR* dir = _wrap->_dir;
+        std::string _dir_path = _wrap->_path;
+        DBG1("FS::Telldir::Work: " << DVAL(_dir_path));
+        _tell_res = telldir(dir);
+        if (_tell_res == -1) SetSyscallError();
+    }
+    virtual void OnOK()
+    {
+        DBG0("FS::Telldir::OnOK: " << DVAL(_dir_path) << DVAL(_tell_res));
+        char telldir_loc[256];
+        sprintf(telldir_loc, "%ld", _tell_res);
```

Comment:
@guymguym I reverted the changes,
I didnt get the your suggestion there, After getting more understating I will implement it in next related PRs

Reply:
sure, no worries, it was mostly a concern for supporting different platforms where the dir pos might be 64 bit, but for now since we use libc for that the type is fixed on long so no real concern there. we can followup if needed on next PR's like you said.